### PR TITLE
update for directory layout and config files

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | yes
| Deprecations?     | no
| Fixed tickets     | #4052 
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

## Description of included changes

This PR includes all changes of the first 2 sections outlined in #4052 (see which points have been checked there):

- Eliminated `app`
- Eliminated `lib`
- Structured `config` files

### CODE REVIEW

**Comment by craigh** [comment]
might be good to stop somewhere in this area and get to a working mergeable state, then a new PR with further changes

**Comment by Guite** [comment]
Only fixing Travis and stopping then.

**Comment by Guite** [comment]
@craigh this PR is ready

File: CHANGELOG-3.0.md [hunk 1]
```diff
@@ -88,10 +88,22 @@
...
     - Removed the `%temp_dir%` parameter. If you need a temporary folder use `sys_get_temp_dir()`.
-    - Removed custom translation system (#4042).
+    - Removed custom translation system (#4042). Use Symfony's translation system directly.
     - Removed use of `admin.png` and replaced by adding icon class to `composer.json` >> `extra/zikula/capabilities/admin/icon: "fas fa-user"`
     - Replaced `LinkContainer` with `ExtensionMenu` for collecting module menus (admin, user, account). See companion docs.
     - Setting `composer.json` >> `extra/zikula/capabilities/admin/url` is no longer supported. Use `extra/zikula/capabilities/admin/route`.
+    - Changes regarding directory layout
+        - The `app/config/` directory has been moved to `config/`.
+        - The `app/Resources/<BundleName>/views/` directory is now located at `templates/bundles/<BundleName>/`.
+        - The `app/Resources/` directory became `Resources/`.
```
Comment (craigh) [inline CHANGELOG-3.0.md]: this part seems odd to me (and `Resources/` is not in the best-practices document). With everything else removed from this, I suspect the last remaining directory is `workflows/`? I recommend we eliminate `Resources/` entirely and only have `worklows/` (like `templates`, `translations` etc)

Comment (Guite) [reply CHANGELOG-3.0.md]: It is from https://github.com/symfony/symfony/blob/master/UPGRADE-4.0.md

Comment (craigh) [reply CHANGELOG-3.0.md]: ok - but is workflows better in `src/worflows/` ? and if so, what is left to put in `Resources/`?

Comment (Guite) [reply CHANGELOG-3.0.md]: Removed `Resources` entirely in https://github.com/zikula/core/commit/9732263c64509b471248b44a15b49fecc60b8a3f

custom workflows can also be added by amending `config/packages/workflows.yaml` or possibly even `config/services_custom.yaml`

Comment (Guite) [reply CHANGELOG-3.0.md]: added `src/workflows/`

**Comment by craigh** [comment]
merge when you're ready.

**Pull request merged**